[SDL3] Build on OmniOS ( Solaris ) fails in SDL_systime.c

Building SDL 3 in OmniOS ( Solaris ) as a preliminary step to build Exult, the build fails as :

```
 [ 89%] Building C object CMakeFiles/SDL3-shared.dir/src/time/unix/SDL_systime.c.o
  /root/SDL3/src/time/unix/SDL_systime.c: In function 'SDL_TimeToDateTime_REAL':
  /root/SDL3/src/time/unix/SDL_systime.c:189:33: error: 'struct tm' has no member named 'tm_gmtoff'
    189 |         dt->utc_offset = (int)tm->tm_gmtoff;
        |                                 ^~
  gmake[2]: *** [CMakeFiles/SDL3-shared.dir/build.make:3454: CMakeFiles/SDL3-shared.dir/src/time/unix/SDL_systime.c.o] Error 1
  gmake[1]: *** [CMakeFiles/Makefile2:94: CMakeFiles/SDL3-shared.dir/all] Error 2
  gmake: *** [Makefile:156: all] Error 2
```

You can find the whole build action as https://github.com/exult/exult/actions/runs/13887660421/job/38854534533.

The GitHub workflow for SDL 3 is simple :
```
git clone https://github.com/libsdl-org/SDL SDL3
cd SDL3
mkdir -p build
cmake -B build -S .
cmake --build build
cmake --install build
```

--- a/src/time/unix/SDL_systime.c
+++ b/src/time/unix/SDL_systime.c
@@ -27,6 +27,7 @@
 #include <langinfo.h>
 #include <sys/time.h>
 #include <time.h>
+#include <unistd.h>
 
 #if !defined(HAVE_CLOCK_GETTIME) && defined(SDL_PLATFORM_APPLE)
 #include <mach/clock.h>
@@ -186,7 +187,23 @@ bool SDL_TimeToDateTime(SDL_Time ticks, SDL_DateTime *dt, bool localTime)
         dt->second = tm->tm_sec;
         dt->nanosecond = ticks % SDL_NS_PER_SECOND;
         dt->day_of_week = tm->tm_wday;
+
+        /* tm_gmtoff wasn't formally standardized until POSIX.1-2024, but practically it has been available on desktop
+         * *nix platforms such as Linux/glibc, FreeBSD, OpenBSD, NetBSD, OSX/macOS, and others since the 1990s.
+         *
+         * The notable exception is Solaris, where the timezone offset must still be retrieved in the strictly POSIX.1-2008
+         * compliant way.
+         */
+#if (_POSIX_VERSION >= 202405L) || (!defined(sun) && !defined(__sun))
         dt->utc_offset = (int)tm->tm_gmtoff;
+#else
+        if (localTime) {
+            tzset();
+            dt->utc_offset = (int)timezone;
+        } else {
+            dt->utc_offset = 0;
+        }
+#endif
 
         return true;
     }

